### PR TITLE
fix(ui): handle special symbols in filenames

### PR DESCRIPTION
**Pull request by rudolfs**
- [x] add tests
- [x] handle backslash `faux\path`

Closes: https://github.com/radicle-dev/radicle-upstream/issues/1021.

### CODE REVIEW

**Comment by xla** [comment]
> Refs: #1021.

Does it not close the issue?

**Comment by rudolfs** [comment]
> Does it not close the issue?

What do you mean @xla?

**Comment by rudolfs** [comment]
> Does it not close the issue?

Switched to `closes`.

File: cypress/integration/project_source_browsing.spec.js [hunk 1]
```diff
@@ -387,6 +387,54 @@ context("project source browsing", () => {
             cy.contains("Binary content").should("exist");
           });
         });
+
+        context("for filenames with special characters", () => {
+          it("does not break", () => {
+            cy.pick("expand-special").click();
+
+            cy.pick("source-tree").contains("-dash-").click();
+            cy.contains("platinum / special / -dash").should("exist");
```
Comment (MeBrei) [inline cypress/integration/project_source_browsing.spec.js:396]:
```suggestion
            cy.contains("platinum / special / -dash-").should("exist");
```

**Review by MeBrei — APPROVED** [review]
⭐

**Review by MeBrei — APPROVED** [review]

File: ui/src/source.ts [hunk 2]
```diff
@@ -264,7 +264,7 @@ const update = (msg: Msg): void => {
               query: {
                 peerId: msg.peerId,
                 revision: msg.revision,
-                path: msg.path,
+                path: encodeURIComponent(msg.path),
```
Comment (xla) [inline ui/src/source.ts:267]: Small change, big gains!

**Review by xla — APPROVED** [review]
🔲 🌝 💁 🏝

**Pull request merged**